src/profile.c

```diff
@@ -213,10 +267,28 @@ void HookedLineOutput(Obj func, char type)
 }
 
 void enterFunction(Obj func)
-{ HookedLineOutput(func, 'I'); }
+{
+    // The recursion depth is incremented before this function
+    // is called, so we pass a '-1' modifier.
+    // If there was already a function at the current
+    // RecursionDepth on the function stack, we want to remove it.
+    CheckLeaveFunctionsAfterLongjmp(-1);
+    PushPlist(VisitedDepths, INTOBJ_INT(GetRecursionDepth()));
+    HookedLineOutput(func, 'I');
+}
 
 void leaveFunction(Obj func)
-{ HookedLineOutput(func, 'O'); }
+{
+    // This > 0 should not be necessary, but it avoids
+    // a crash if the function stack has become badly
+    // corrupted.
```

Comment:
Honestly, I can't think how this list could end up empty. However, if it did, the best thing to do would be to carry on -- it might lead to a slightly weird profile.

Reply:
I've just thought how it could get empty -- starting a profile in a function and returning from that function.